api/consensus.go

```diff
@@ -25,25 +35,69 @@ func (srv *Server) consensusStatusHandler(w http.ResponseWriter, req *http.Reque
 		}
 	}
 
-	writeJSON(w, ConsensusInfo{
-		srv.blockchainHeight,
+	writeJSON(w, ConsensusSetStatus{
+		types.BlockHeight(srv.blockchainHeight),
 		srv.currentBlock.ID(),
 		currentTarget,
 	})
 }
 
-// consensusSynchronizeHandler handles the API call asking for the consensus to
-// synchronize with other peers.
-func (srv *Server) consensusSynchronizeHandler(w http.ResponseWriter, req *http.Request) {
+// consensusHandler handles the API calls to /consensus.
+func (srv *Server) consensusHandler(w http.ResponseWriter, req *http.Request) {
+	lockID := srv.mu.RLock()
+	defer srv.mu.RUnlock(lockID)
+
+	if req.Method == "" || req.Method == "GET" {
+		srv.consensusHandlerGET(w, req)
+		return
+	}
```

Comment:
if you're planning on doing this for the rest of the API functions, you should write a wrapper function.
```go
func methodWrapper(fn http.HandleFunc, method string) http.HandleFunc {
    return func(w http.ResponseWriter, req *http.Request) {
        if req.Method != method {
            writeError(w, "bad method; use " + method)
            return
        }
        fn(w, req)
    }
}
```